src/rafter_consensus_fsm.erl

```diff
@@ -841,25 +823,39 @@ become_candidate(#state{term=CurrentTerm, me=Me}=State0) ->
     _ = request_votes(State3),
     State3.
 
-become_leader(#state{me=Me, term=Term, init_config=InitConfig}=State) ->
-    NewState = State#state{leader=Me,
+become_leader(#state{me=Me, term=Term, config=Config, init_config=InitConfig, timer=Timer}=State) ->
```

Comment:
Timer is not modified and is just put back in the new state. Please remove `timer=Timer` here and below.